feature/src/main/java/com/terning/feature/mypage/profileedit/ProfileEditRoute.kt

```diff
@@ -0,0 +1,177 @@
+package com.terning.feature.mypage.profileedit
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.flowWithLifecycle
+import com.terning.core.designsystem.component.bottomsheet.SignUpBottomSheet
+import com.terning.core.designsystem.component.button.RectangleButton
+import com.terning.core.designsystem.component.item.ProfileWithPlusButton
+import com.terning.core.designsystem.component.textfield.NameTextField
+import com.terning.core.designsystem.component.topappbar.BackButtonTopAppBar
+import com.terning.core.designsystem.theme.Grey500
+import com.terning.core.designsystem.theme.TerningPointTheme
+import com.terning.core.designsystem.theme.TerningTheme
+import com.terning.core.extension.addFocusCleaner
+import com.terning.core.extension.noRippleClickable
+import com.terning.feature.R
+
+@Composable
+fun ProfileEditRoute(
+    navigateUp: () -> Unit,
+    viewModel: ProfileEditViewModel = hiltViewModel(),
+    initialName: String = "",
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    LaunchedEffect(key1 = true) {
+        viewModel.updateName(initialName)
+    }
+
+    if (state.showBottomSheet) {
+        SignUpBottomSheet(
+            onDismiss = { viewModel.updateBottomSheet(false) },
+            onSaveClick = { index ->
+                viewModel.updateBottomSheet(false)
+                viewModel.updateCharacter(index)
+            },
+            initialSelectedOption = state.character
+        )
+    }
+
+    LaunchedEffect(viewModel.sideEffects, lifecycleOwner) {
+        viewModel.sideEffects.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
+            .collect { sideEffect ->
+                when (sideEffect) {
+                    is ProfileEditSideEffect.NavigateUp -> navigateUp()
+                }
+            }
+    }
+
+    ProfileEditScreen(
+        profileEditState = state,
+        onProfileEditClick = { isVisible ->
+            viewModel.updateBottomSheet(isVisible)
+        },
+        onInputChange = { editName ->
+            viewModel.isInputValid(editName)
+        },
```

Comment:
이렇게 하나의 함수만 사용하고, 람다 내 변수의 타입이 함수 매개변수 타입과 같으면 참조 연산자 (::)를 사용할 수 있답니다!
```suggestion
        onInputChange = viewModel::isInputValid,
```

취향따라 사용하면 되지만 저는 이게 더 깔끔해보이더라요ㅎㅎ